File: react-native/client/ios/Records/ActionResultRecord.swift
import CoinbaseWalletSDK
import ExpoModulesCore
import Foundation

struct ActionResultRecord : Record {
    @Field
    var result: String?

    @Field
    var errorMessage: String?

    @Field
    var errorCode: Int?
}

extension ActionResult {
    var asRecord: ActionResultRecord.Dict {
        let record = ActionResultRecord()

        switch self {
        case .success(let value):
            record.result = value.rawValue
        case .failure(let error):
            record.errorCode = error.code
            record.errorMessage = error.message
        }

        return record.toDictionary()
    }
}

File: react-native/android/src/main/java/expo/modules/coinbasewalletsdkexpo/records/ActionResultRecord.kt
package expo.modules.coinbasewalletsdkexpo.records

import com.coinbase.android.nativesdk.message.response.ActionResult
import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class ActionResultRecord : Record {
    @Field
    var result: String? = null

    @Field
    var errorMessage: String? = null

    @Field
    var errorCode: Int? = null
}

val ActionResult.asRecord: ActionResultRecord
    get() {
        val record = ActionResultRecord()
        when (this) {
            is ActionResult.Result -> record.result = value
            is ActionResult.Error -> {
                record.errorCode = code.toInt()
                record.errorMessage = message
            }
        }

        return record
    }